### PR TITLE
AD6676 driver update

### PR DESCRIPTION
**Pull request by adrimbarean**
Update charge pump initialization routine to include corner case specified in the datasheet.
Fix reserved bit mask of a register.

### CODE REVIEW

**Comment by adrimbarean** [comment]
V0 -> V1:
- improve if coding
- remove `CLKSYN_R_DIV_RESERVED` change because the way it is used might be different than what the name implies: it is used in the register initialization routine as a initialization value rather than as a mask 
https://github.com/analogdevicesinc/no-OS/blob/master/drivers/ad6676/ad6676.c#L373

**Comment by commodo** [comment]
if this needs fixing here, it may need fixing in the Linux driver
@mhennerich can you take a look as well ?

code looks similar to Linux; so if this patch is fine here, we could do the same for the Linux driver

File: drivers/ad6676/ad6676.h [hunk 1]
```diff
@@ -209,7 +209,7 @@
  */
 
 #define R_DIV(x)			((x) << 6);
-#define CLKSYN_R_DIV_RESERVED		0x3D
+#define CLKSYN_R_DIV_RESERVED		0x33
```
[the remark this answers is not in the document]
Comment (mhennerich) [reply drivers/ad6676/ad6676.h]: According to the latest datasheet - the reserved bits are following:
#define CLKSYN_R_DIV_RESERVED 0x31

The driver should be updated this way.
Furthermore we need to see if the currently set bits are required or not.

In any case if we fix or update no-OS drivers with Linux driver counterparts, they *must* be updated, too.

**Comment by adrimbarean** [comment]
V1 -> V2:
- fix CLKSYN_R_DIV_RESERVED reserved define
- update CLKSYN_R_DIV register description with appropriate defines
- remove stray semicolon

**Comment by adrimbarean** [comment]
V2 -> V3:
- squash related commits together
- preserve functionality by adding the new defines to the initialization

**Comment by commodo** [comment]
After a bit of back-n-forth on this idea, the conclusion came about to leave the bits set as before but to at least break the  CLKSYN_R_DIV_RESERVED into the correct bit fields.

If no objections, will merge this on Monday.
No sense in rushing, since we took our time with this.

**Comment by commodo** [comment]
Ah, found this one that fits here.

[LGTM](https://qph.fs.quoracdn.net/main-qimg-a2ded02032b21e304f9208d12ff1ccbb)

**Pull request merged**